### PR TITLE
Simon/floating avatars remove animation

### PR DESCRIPTION
**Pull request by ShimonScheff**
Floating Avatars can be static

### CODE REVIEW

File: projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.component.ts [hunk 1]
```diff
@@ -71,22 +81,56 @@ export class FloatingAvatarsComponent implements OnInit, OnDestroy {
...
+      const ballData = this.staticAvatarsLocation[i] ? this.staticAvatarsLocation[i] : this.invisibleAvatar;
+
+      particle = new Ball(
+        ballData.ballSize,
+        this.avatarImages[i],
+      );
+      particle.x = ballData.x * this.canvasDimension.width;
+      particle.y = ballData.y * this.canvasDimension.height;
+      particle.vx = 0;
+      particle.vy = 0;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.component.ts]: this looks like a lot of duplicated code, why do we need this?

File: projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.component.ts [hunk 2]
```diff
@@ -71,22 +81,56 @@ export class FloatingAvatarsComponent implements OnInit, OnDestroy {
...
+    }
+    if (this.centerAvatarImage) {
+      let particle: Ball;
+      particle = new Ball(90, this.centerAvatarImage);
+      particle.x = this.canvasDimension.width / 2;
+      particle.y = this.canvasDimension.height / 2;
+      particle.vx = 1;
+      particle.vy = 1;
+      particle.isCenter = true;
+      this.particles.push(particle);
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.component.ts]: didn't we have this somewhere else?

Comment (ShimonScheff) [reply projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.component.ts]: fixed

File: projects/ui-framework/src/lib/mock.const.ts [hunk 3]
```diff
@@ -1099,6 +1100,51 @@ export const mockAvatar = () =>
     'women',
   ])}/${randomNumber(0, 99)}.jpg`;
 
+export const mockStaticAvatarsLocation = (): StaticAvatarLocation[] => {
```
Comment (galloween) [inline projects/ui-framework/src/lib/mock.const.ts]: this should be in floating-avatars.mock.ts - as its used only in floating avatars.
the global mocks file is for stuff that is used a lot, not specific  component mocks

Comment (ShimonScheff) [reply projects/ui-framework/src/lib/mock.const.ts]: Good to Know :)
Fixed

File: projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.interface.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export interface StaticAvatarLocation {
+  ballSize: number;
```
Comment (galloween) [inline projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.interface.ts]: lets call it avatarSize (or even just size ;)
avatars dont have balls!

File: projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.stories.ts [hunk 5]
```diff
@@ -21,10 +24,22 @@ const avatarImagesMock: string[] = Array.from(Array(avatarNum), (_, i) => {
...
+
 const template = `
-  <b-floating-avatars [avatarImages]="avatarImages"
+  <b-floating-avatars *ngIf="animation" [avatarImages]="avatarImages"
+                      [centerAvatarImage]="centerAvatarImage"
+                      [speed]="speed"
+                      [animation]="animation"
+                      >
+  </b-floating-avatars>
+  <b-floating-avatars *ngIf="!animation" [avatarImages]="avatarImages"
```
[the remark this answers is not in the document]
Comment (ShimonScheff) [reply projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.stories.ts]: It is just for the story book.

File: projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.component.ts [hunk 6]
```diff
@@ -71,32 +75,49 @@ export class FloatingAvatarsComponent implements OnInit, OnDestroy {
...
       width: this.hostRef.nativeElement.clientWidth,
-      height: this.hostRef.nativeElement.clientHeight
+      height: this.hostRef.nativeElement.clientHeight,
     };
     this.canvasEl.width = this.canvasDimension.width;
     this.canvasEl.height = this.canvasDimension.height;
   }
 
   private setScene(): void {
+    if (!this.animation) {
```
[the remark this answers is not in the document]
Comment (ShimonScheff) [reply projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.component.ts]: The default is an empty array

File: projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.stories.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ShimonScheff) [reply projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.stories.ts]: You only need 1 instance when using the component

File: projects/ui-framework/src/lib/mock.const.ts [hunk 7]
```diff
@@ -6,6 +6,7 @@ import {
   capitalize,
 } from './services/utils/functional-utils';
 import { Icons } from './icons/icons.enum';
+import { StaticAvatarLocation } from './eye-candy/floating-avatars/floating-avatars.interface';
```
[the remark this answers is not in the document]
Comment (ShimonScheff) [reply projects/ui-framework/src/lib/mock.const.ts]: fixed

File: projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.interface.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
+export interface StaticAvatarLocation {
+  avatarSize: number;
+  x: number;
+  y: number;
+}
+
+export const staticAvatarLocationDesktopDefault: StaticAvatarLocation[] = [
```
Comment (galloween) [inline projects/ui-framework/src/lib/eye-candy/floating-avatars/floating-avatars.interface.ts]: the consts now belong to floating-avatars.const.ts :)

**Comment by galloween** [comment]
@ShimonScheff 
You can ignore my comments. I took your version and extended on it. It seems to be working fine at last. Take a look: https://github.com/hibobio/bob-style/pull/1387
It still needs better location for mobile (now you can test mobile view by resizing the window)

**Pull request merged**